src/track/annotation/AnnotationTrack.tsx

```diff
@@ -238,6 +242,34 @@ export class AnnotationTrack extends TrackObject<AnnotationTrackModel, Annotatio
                 annotation.name.x = 5
                 annotation.name.relativeX = Math.max(annotation.gene.relativeX, 0);
 
+                //VALIS-8: prevent collision between annotation names
+                let annotationNameWidth = annotation.name.getComputedWidth();
+                if (annotation.name.visible && annotationNameWidth) {
+                    for (let name of createdAnnotationNames) {
+                        if (annotation.name.y == name.y && gene.name != name.geneName) {
+                            // check gene's name to prevent multiple collision checking
+                            // on the same gene that is rendered multiple times in different tile.payload
```

Comment:
I'm afraid I find this comment confusing. Do you mean that a gene's label can overlap with a lot of other labels so you have to check more than just the ones on the immediate left and right? Also why would a label be rendered multiple times?

Reply:
We change `annotation.name.relativeX` to shift gene's label to the right and if that crosses the boundary of the current tile block then it's queued to the next tile and rendered again. This occurs only when we increase `annotation.name.relativeX` and we can simply avoid it by ignoring genes that are already shifted.

Reply:
Updated comments.

Reply:
Gene labels don't fade out anymore.